wiki/start/a-simple-game.md

```diff
@@ -5,605 +5,1091 @@ redirect_from:
+        bucketSprite.setX(touchPos.x); //Change the horizontal position of the bucket
+    }
+}
 ```
 
-The first line tells the `SpriteBatch` to use the coordinate system specified by the camera. As stated earlier, this is done with something called a matrix, to be more specific, a projection matrix. The `camera.combined` field is such a matrix. From there on the `SpriteBatch` will render everything in the coordinate system described earlier.
+This converts the window coordinates to coordinates in our world space. This code actually supports mobile devices as well, however you should read about [some other input features](/wiki/input/event-handling) that libGDX provides you. Run the game and click the screen to move the bucket.
 
-Next we tell the `SpriteBatch` to start a new batch. Why do we need this and what is a batch? OpenGL hates nothing more than telling it about individual images. It wants to be told about as many images to render as possible at once.
+## Game Logic
+The player can move left and right now, but they can go completely off the screen. We need to prevent the player from doing that. Remember that the left side of the screen starts at 0.
+This code detects if the bucket goes too far left. If it does, it snaps its position at the farthest it's allowed to go. Add the following line to the logic method
 
-The `SpriteBatch` class helps make OpenGL happy. It will record all drawing commands in between `SpriteBatch.begin()` and `SpriteBatch.end()`. Once we call `SpriteBatch.end()` it will submit all drawing requests we made at once, speeding up rendering quite a bit. This all might look cumbersome in the beginning, but it is what makes the difference between rendering 500 sprites at 60 frames per second and rendering 100 sprites at 20 frames per second.
+```java
+private void logic() {
+    if (bucketSprite.getX() < 0) bucketSprite.setX(0); //If x is less than 0, reset to 0
+}
+```
 
-### Making the Bucket Move (Touch/Mouse)
-Time to let the user control the bucket. Earlier we said we'd allow the user to drag the bucket. Let's make things a little bit easier. If the user touches the screen (or presses a mouse button), we want the bucket to center around that position horizontally. Adding the following code to the bottom of the `render()` method will do this:
+Let's do the other side.
 
 ```java
-   if(Gdx.input.isTouched()) {
-      Vector3 touchPos = new Vector3();
-      touchPos.set(Gdx.input.getX(), Gdx.input.getY(), 0);
-      camera.unproject(touchPos);
-      bucket.x = touchPos.x - 64 / 2;
-   }
+private void logic() {
+    //Store the worldWidth and worldHeight as local variables for brevity
+    float worldWidth = viewport.getWorldWidth();
+    float worldHeight = viewport.getWorldHeight();
+
+    if (bucketSprite.getX() < 0) bucketSprite.setX(0);
+    else if (bucketSprite.getX() > worldWidth) //If x is greater than world width
+        bucketSprite.setX(worldWidth); //Reset x to world width
```

Comment:
If we're using Lyze' suggestion of making `4f` a declared value above this block, then it would be `+ (speed * delta)` and `- (speed * delta)`, or speed could already incorporate delta if it is declared locally. There's some nice symmetry there.